_posts/2020-07-30-axon-kompleksowe-testowanie-aplikacji.md

```diff
@@ -0,0 +1,386 @@
+---
+layout:    post
+title:     "Axon - Kompleksowe testowanie aplikacji"
+date:      2020-07-30 08:00:00 +0100
+published: true
+author:    mkociszewski
+tags:
+    - java
+    - microservices
+    - axon
+    - event-sourcing
+    - spring-boot
+    - tests
+---
+
+Powszechnie wiadomo, że kod dobrze pokryty testami jest dużo bardziej podatny na rozwój - wszak nie musimy obawiać się, że nasza zmiana coś zepsuje, a my się o tym nie dowiemy.
```

Comment:
testy nigdy nie powiedzą, że w czymś nie ma błędu; testy jedynie wskażą, że nie wróciły znane błędy lub że błędy są ;-)

Reply:
A idealnie napisane testy nie wskażą mi, że mój kawałek kodu działa dokładnie tak jak sobie wymyśliłem (czyli bezbłędnie)? Oczywiście mogę zapomnieć jakiegoś corner case'a, ale to wtedy nie są `idealnie` napisane testy :smile:.
Takowe testy mogą istnieć niestety tylko w hermetycznie zamkniętym, wyidealizowanym środowisku (czyli nie istnieją), więc poprawiam. :)

Reply:
:D